### PR TITLE
freeradius3: update version 3.2.5

### PR DESCRIPTION
**Pull request by esaaprillia**
Maintainer:
Compile tested: aarch64
Run tested: aarch64/ openwrt 23.05.3

Description:

https://github.com/FreeRADIUS/freeradius-server/releases/tag/release_3_2_5

update freeradius version 3.2.5

refresh patch

```
root@OpenWrt:~/mosdns# radiusd -X
FreeRADIUS Version 3.2.5
Copyright (C) 1999-2023 The FreeRADIUS server project and contributors
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE
You may redistribute copies of FreeRADIUS under the terms of the
GNU General Public License
For more information about these matters, see the file named COPYRIGHT
(TLS) Failed loading legacy provider
root@OpenWrt:~#
root@OpenWrt:~#
root@OpenWrt:~# opkg install libopenssl-legacy
Installing libopenssl-legacy (3.0.14-2) to root...
Downloading https://downloads.openwrt.org/releases/23.05.3/packages/aarch64_generic/base/libopenssl-legacy_3.0.14-2_aarch64_generic.ipk
Installing libopenssl-conf (3.0.14-2) to root...
Downloading https://downloads.openwrt.org/releases/23.05.3/packages/aarch64_generic/base/libopenssl-conf_3.0.14-2_aarch64_generic.ipk
Configuring libopenssl-conf.
Generating engines.cnf
Generating providers.cnf
Configuring libopenssl-legacy.
Generating engines.cnf
Generating providers.cnf
Enabling legacy
root@OpenWrt:~/mosdns# radiusd -X
FreeRADIUS Version 3.2.5
Copyright (C) 1999-2023 The FreeRADIUS server project and contributors
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE
You may redistribute copies of FreeRADIUS under the terms of the
GNU General Public License
For more information about these matters, see the file named COPYRIGHT
Starting - reading configuration files ...
including dictionary file /usr/share/freeradius3/dictionary
including dictionary file /etc/freeradius3/dictionary
including configuration file /etc/freeradius3/radiusd.conf
including configuration file /etc/freeradius3/proxy.conf
including configuration file /etc/freeradius3/clients.conf
including files in directory /etc/freeradius3/mods-enabled/
including configuration file /etc/freeradius3/mods-enabled/always
including configuration file /etc/freeradius3/mods-enabled/attr_filter
including configuration file /etc/freeradius3/mods-enabled/chap
including configuration file /etc/freeradius3/mods-enabled/detail
including configuration file /etc/freeradius3/mods-enabled/digest
including configuration file /etc/freeradius3/mods-enabled/eap
including configuration file /etc/freeradius3/mods-enabled/mschap
including configuration file /etc/freeradius3/mods-enabled/exec
including configuration file /etc/freeradius3/mods-enabled/expiration
including configuration file /etc/freeradius3/mods-enabled/expr
including configuration file /etc/freeradius3/mods-enabled/files
including configuration file /etc/freeradius3/mods-enabled/logintime
including configuration file /etc/freeradius3/mods-enabled/pap
including configuration file /etc/freeradius3/mods-enabled/preprocess
including configuration file /etc/freeradius3/mods-enabled/radutmp
including configuration file /etc/freeradius3/mods-enabled/sradutmp
including configuration file /etc/freeradius3/mods-enabled/realm
including configuration file /etc/freeradius3/mods-enabled/unix
including configuration file /etc/freeradius3/mods-enabled/passwd
including files in directory /etc/freeradius3/policy.d/
including configuration file /etc/freeradius3/policy.d/accounting
including configuration file /etc/freeradius3/policy.d/filter
including configuration file /etc/freeradius3/policy.d/eap
including files in directory /etc/freeradius3/sites-enabled/
including configuration file /etc/freeradius3/sites-enabled/default
including configuration file /etc/freeradius3/sites-enabled/inner-tunnel
main {
 security {
        allow_core_dumps = no
 }
        name = "radiusd"
        prefix = "/usr"
        localstatedir = "/var"
        logdir = "/var/log"
        run_dir = "/var/run/radiusd"
}
main {
        name = "radiusd"
        prefix = "/usr"
        localstatedir = "/var"
        sbindir = "/usr/sbin"
        logdir = "/var/log"
        run_dir = "/var/run/radiusd"
        libdir = "/usr/lib/freeradius3"
        radacctdir = "/var/db/radacct"
        hostname_lookups = no
        max_request_time = 30
        proxy_dedup_window = 1
        cleanup_delay = 5
        max_requests = 16384
        max_fds = 512
        postauth_client_lost = no
        pidfile = "/var/run/radiusd/radiusd.pid"
        checkrad = "/usr/sbin/checkrad"
        debug_level = 0
        proxy_requests = yes
 log {
        stripped_names = no
        auth = no
        auth_badpass = no
        auth_goodpass = no
        colourise = yes
        msg_denied = "You are already logged in - access denied"
 }
 resources {
 }
 security {
        max_attributes = 200
        reject_delay = 1.000000
        status_server = yes
        require_message_authenticator = "auto"
        limit_proxy_state = "auto"
 }
}
radiusd: #### Loading Realms and Home Servers ####
 proxy server {
        retry_delay = 5
        retry_count = 3
        default_fallback = no
        dead_time = 120
        wake_all_if_all_dead = no
 }
 home_server localhost {
        nonblock = no
        ipaddr = 127.0.0.1
        port = 1812
        type = "auth"
        secret = <<< secret >>>
        response_window = 20.000000
        response_timeouts = 1
        max_outstanding = 65536
        zombie_period = 40
        status_check = "status-server"
        ping_interval = 30
        check_interval = 30
        check_timeout = 4
        num_answers_to_alive = 3
        revive_interval = 120
  limit {
        max_connections = 16
        max_requests = 0
        lifetime = 0
        idle_timeout = 0
  }
  coa {
        irt = 2
        mrt = 16
        mrc = 5
        mrd = 30
  }
 }
 home_server_pool my_auth_failover {
        type = fail-over
        home_server = localhost
 }
 realm example.com {
        auth_pool = my_auth_failover
 }
 realm LOCAL {
 }
radiusd: #### Loading Clients ####
 client localhost {
        ipaddr = 127.0.0.1
        secret = <<< secret >>>
        nas_type = "other"
        proto = "*"
  limit {
        max_connections = 16
        lifetime = 0
        idle_timeout = 30
  }
 }
 client localhost_ipv6 {
        ipv6addr = ::1
        secret = <<< secret >>>
  limit {
        max_connections = 16
        lifetime = 0
        idle_timeout = 30
  }
 }
Debugger not attached
 # Creating Auth-Type = mschap
 # Creating Auth-Type = digest
 # Creating Auth-Type = eap
 # Creating Auth-Type = PAP
 # Creating Auth-Type = CHAP
 # Creating Auth-Type = MS-CHAP
 # Creating Autz-Type = New-TLS-Connection
radiusd: #### Instantiating modules ####
 modules {
  # Loaded module rlm_always
  # Loading module "reject" from file /etc/freeradius3/mods-enabled/always
  always reject {
        rcode = "reject"
        simulcount = 0
        mpp = no
  }
  # Loading module "fail" from file /etc/freeradius3/mods-enabled/always
  always fail {
        rcode = "fail"
        simulcount = 0
        mpp = no
  }
  # Loading module "ok" from file /etc/freeradius3/mods-enabled/always
  always ok {
        rcode = "ok"
        simulcount = 0
        mpp = no
  }
  # Loading module "handled" from file /etc/freeradius3/mods-enabled/always
  always handled {
        rcode = "handled"
        simulcount = 0
        mpp = no
  }
  # Loading module "invalid" from file /etc/freeradius3/mods-enabled/always
  always invalid {
        rcode = "invalid"
        simulcount = 0
        mpp = no
  }
  # Loading module "userlock" from file /etc/freeradius3/mods-enabled/always
  always userlock {
        rcode = "userlock"
        simulcount = 0
        mpp = no
  }
  # Loading module "notfound" from file /etc/freeradius3/mods-enabled/always
  always notfound {
        rcode = "notfound"
        simulcount = 0
        mpp = no
  }
  # Loading module "noop" from file /etc/freeradius3/mods-enabled/always
  always noop {
        rcode = "noop"
        simulcount = 0
        mpp = no
  }
  # Loading module "updated" from file /etc/freeradius3/mods-enabled/always
  always updated {
        rcode = "updated"
        simulcount = 0
        mpp = no
  }
  # Loaded module rlm_attr_filter
  # Loading module "attr_filter.post-proxy" from file /etc/freeradius3/mods-enabled/attr_filter
  attr_filter attr_filter.post-proxy {
        filename = "/etc/freeradius3/mods-config/attr_filter/post-proxy"
        key = "%{Realm}"
        relaxed = no
  }
  # Loading module "attr_filter.pre-proxy" from file /etc/freeradius3/mods-enabled/attr_filter
  attr_filter attr_filter.pre-proxy {
        filename = "/etc/freeradius3/mods-config/attr_filter/pre-proxy"
        key = "%{Realm}"
        relaxed = no
  }
  # Loading module "attr_filter.access_reject" from file /etc/freeradius3/mods-enabled/attr_filter
  attr_filter attr_filter.access_reject {
        filename = "/etc/freeradius3/mods-config/attr_filter/access_reject"
        key = "%{User-Name}"
        relaxed = no
  }
  # Loading module "attr_filter.access_challenge" from file /etc/freeradius3/mods-enabled/attr_filter
  attr_filter attr_filter.access_challenge {
        filename = "/etc/freeradius3/mods-config/attr_filter/access_challenge"
        key = "%{User-Name}"
        relaxed = no
  }
  # Loading module "attr_filter.accounting_response" from file /etc/freeradius3/mods-enabled/attr_filter
  attr_filter attr_filter.accounting_response {
        filename = "/etc/freeradius3/mods-config/attr_filter/accounting_response"
        key = "%{User-Name}"
        relaxed = no
  }
  # Loading module "attr_filter.coa" from file /etc/freeradius3/mods-enabled/attr_filter
  attr_filter attr_filter.coa {
        filename = "/etc/freeradius3/mods-config/attr_filter/coa"
        key = "%{User-Name}"
        relaxed = no
  }
  # Loaded module rlm_chap
  # Loading module "chap" from file /etc/freeradius3/mods-enabled/chap
  # Loaded module rlm_detail
  # Loading module "detail" from file /etc/freeradius3/mods-enabled/detail
  detail {
        filename = "/var/db/radacct/%{%{Packet-Src-IP-Address}:-%{Packet-Src-IPv6-Address}}/detail-%Y%m%d"
        header = "%t"
        permissions = 384
        locking = no
        dates_as_integer = no
        escape_filenames = no
        log_packet_header = no
  }
  # Loaded module rlm_digest
  # Loading module "digest" from file /etc/freeradius3/mods-enabled/digest
  # Loaded module rlm_eap
  # Loading module "eap" from file /etc/freeradius3/mods-enabled/eap
  eap {
        default_eap_type = "md5"
        timer_expire = 60
        max_eap_type = 52
        ignore_unknown_eap_types = no
        cisco_accounting_username_bug = no
        max_sessions = 16384
        dedup_key = ""
  }
  # Loaded module rlm_mschap
  # Loading module "mschap" from file /etc/freeradius3/mods-enabled/mschap
  mschap {
        use_mppe = yes
        require_encryption = no
        require_strong = no
        with_ntdomain_hack = yes
   passchange {
   }
        allow_retry = yes
        winbind_retry_with_normalised_username = no
  }
  # Loaded module rlm_exec
  # Loading module "exec" from file /etc/freeradius3/mods-enabled/exec
  exec {
        wait = no
        input_pairs = "request"
        shell_escape = yes
        timeout = 10
  }
  # Loaded module rlm_expiration
  # Loading module "expiration" from file /etc/freeradius3/mods-enabled/expiration
  # Loaded module rlm_expr
  # Loading module "expr" from file /etc/freeradius3/mods-enabled/expr
  expr {
        safe_characters = "@abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789.-_: /äéöüàâæçèéêëîïôœùûüaÿÄÉÖÜßÀÂÆÇÈÉÊËÎÏÔŒÙÛÜŸ"
  }
  # Loaded module rlm_files
  # Loading module "files" from file /etc/freeradius3/mods-enabled/files
  files {
        filename = "/etc/freeradius3/mods-config/files/authorize"
        acctusersfile = "/etc/freeradius3/mods-config/files/accounting"
        preproxy_usersfile = "/etc/freeradius3/mods-config/files/pre-proxy"
  }
  # Loaded module rlm_logintime
  # Loading module "logintime" from file /etc/freeradius3/mods-enabled/logintime
  logintime {
        minimum_timeout = 60
  }
  # Loaded module rlm_pap
  # Loading module "pap" from file /etc/freeradius3/mods-enabled/pap
  pap {
        normalise = yes
  }
  # Loaded module rlm_preprocess
  # Loading module "preprocess" from file /etc/freeradius3/mods-enabled/preprocess
  preprocess {
        huntgroups = "/etc/freeradius3/mods-config/preprocess/huntgroups"
        hints = "/etc/freeradius3/mods-config/preprocess/hints"
        with_ascend_hack = no
        ascend_channels_per_line = 23
        with_ntdomain_hack = no
        with_specialix_jetstream_hack = no
        with_cisco_vsa_hack = no
        with_alvarion_vsa_hack = no
  }
  # Loaded module rlm_radutmp
  # Loading module "radutmp" from file /etc/freeradius3/mods-enabled/radutmp
  radutmp {
        filename = "/var/log/radutmp"
        username = "%{User-Name}"
        case_sensitive = yes
        check_with_nas = yes
        permissions = 384
        caller_id = yes
  }
  # Loading module "sradutmp" from file /etc/freeradius3/mods-enabled/sradutmp
  radutmp sradutmp {
        filename = "/var/log/sradutmp"
        username = "%{User-Name}"
        case_sensitive = yes
        check_with_nas = yes
        permissions = 420
        caller_id = no
  }
  # Loaded module rlm_realm
  # Loading module "IPASS" from file /etc/freeradius3/mods-enabled/realm
  realm IPASS {
        format = "prefix"
        delimiter = "/"
        ignore_default = no
        ignore_null = no
  }
  # Loading module "suffix" from file /etc/freeradius3/mods-enabled/realm
  realm suffix {
        format = "suffix"
        delimiter = "@"
        ignore_default = no
        ignore_null = no
  }
  # Loading module "bangpath" from file /etc/freeradius3/mods-enabled/realm
  realm bangpath {
        format = "prefix"
        delimiter = "!"
        ignore_default = no
        ignore_null = no
  }
  # Loading module "realmpercent" from file /etc/freeradius3/mods-enabled/realm
  realm realmpercent {
        format = "suffix"
        delimiter = "%"
        ignore_default = no
        ignore_null = no
  }
  # Loading module "ntdomain" from file /etc/freeradius3/mods-enabled/realm
  realm ntdomain {
        format = "prefix"
        delimiter = "\"
        ignore_default = no
        ignore_null = no
  }
  # Loaded module rlm_unix
  # Loading module "unix" from file /etc/freeradius3/mods-enabled/unix
  unix {
        radwtmp = "/var/log/radwtmp"
  }
Creating attribute Unix-Group
  # Loaded module rlm_passwd
  # Loading module "etc_passwd" from file /etc/freeradius3/mods-enabled/passwd
  passwd etc_passwd {
        filename = "/etc/passwd"
        format = "*User-Name:Crypt-Password:"
        delimiter = ":"
        ignore_nislike = no
        ignore_empty = yes
        allow_multiple_keys = no
        hash_size = 100
  }
  instantiate {
  }
  # Instantiating module "reject" from file /etc/freeradius3/mods-enabled/always
  # Instantiating module "fail" from file /etc/freeradius3/mods-enabled/always
  # Instantiating module "ok" from file /etc/freeradius3/mods-enabled/always
  # Instantiating module "handled" from file /etc/freeradius3/mods-enabled/always
  # Instantiating module "invalid" from file /etc/freeradius3/mods-enabled/always
  # Instantiating module "userlock" from file /etc/freeradius3/mods-enabled/always
  # Instantiating module "notfound" from file /etc/freeradius3/mods-enabled/always
  # Instantiating module "noop" from file /etc/freeradius3/mods-enabled/always
  # Instantiating module "updated" from file /etc/freeradius3/mods-enabled/always
  # Instantiating module "attr_filter.post-proxy" from file /etc/freeradius3/mods-enabled/attr_filter
reading pairlist file /etc/freeradius3/mods-config/attr_filter/post-proxy
  # Instantiating module "attr_filter.pre-proxy" from file /etc/freeradius3/mods-enabled/attr_filter
reading pairlist file /etc/freeradius3/mods-config/attr_filter/pre-proxy
  # Instantiating module "attr_filter.access_reject" from file /etc/freeradius3/mods-enabled/attr_filter
reading pairlist file /etc/freeradius3/mods-config/attr_filter/access_reject
  # Instantiating module "attr_filter.access_challenge" from file /etc/freeradius3/mods-enabled/attr_filter
reading pairlist file /etc/freeradius3/mods-config/attr_filter/access_challenge
  # Instantiating module "attr_filter.accounting_response" from file /etc/freeradius3/mods-enabled/attr_filter
reading pairlist file /etc/freeradius3/mods-config/attr_filter/accounting_response
  # Instantiating module "attr_filter.coa" from file /etc/freeradius3/mods-enabled/attr_filter
reading pairlist file /etc/freeradius3/mods-config/attr_filter/coa
  # Instantiating module "detail" from file /etc/freeradius3/mods-enabled/detail
  # Instantiating module "eap" from file /etc/freeradius3/mods-enabled/eap
   # Linked to sub-module rlm_eap_md5
   # Linked to sub-module rlm_eap_gtc
   gtc {
        challenge = "Password: "
        auth_type = "PAP"
   }
   # Linked to sub-module rlm_eap_tls
   tls {
        tls = "tls-common"
   }
   tls-config tls-common {
        verify_depth = 0
        ca_path = "/etc/freeradius3/certs"
        pem_file_type = yes
        private_key_file = "/etc/freeradius3/certs/server.pem"
        certificate_file = "/etc/freeradius3/certs/server.pem"
        ca_file = "/etc/freeradius3/certs/ca.pem"
        private_key_password = <<< secret >>>
        fragment_size = 1024
        include_length = yes
        auto_chain = yes
        check_crl = no
        check_all_crl = no
        ca_path_reload_interval = 0
        cipher_list = "DEFAULT"
        cipher_server_preference = no
        reject_unknown_intermediate_ca = no
        ecdh_curve = ""
        tls_max_version = "1.2"
        tls_min_version = "1.2"
    cache {
        enable = no
        lifetime = 24
        max_entries = 255
    }
    verify {
        skip_if_ocsp_ok = no
    }
    ocsp {
        enable = no
        override_cert_url = yes
        url = "http://127.0.0.1/ocsp/"
        use_nonce = yes
        timeout = 0
        softfail = no
    }
   }
   # Linked to sub-module rlm_eap_ttls
   ttls {
        tls = "tls-common"
        default_eap_type = "md5"
        copy_request_to_tunnel = no
        use_tunneled_reply = no
        virtual_server = "inner-tunnel"
        include_length = yes
        require_client_cert = no
   }
tls: Using cached TLS configuration from previous invocation
   # Linked to sub-module rlm_eap_peap
   peap {
        tls = "tls-common"
        default_eap_type = "mschapv2"
        copy_request_to_tunnel = no
        use_tunneled_reply = no
        proxy_tunneled_request_as_eap = yes
        virtual_server = "inner-tunnel"
        soh = no
        require_client_cert = no
   }
tls: Using cached TLS configuration from previous invocation
   # Linked to sub-module rlm_eap_mschapv2
   mschapv2 {
        with_ntdomain_hack = no
        send_error = no
   }
  # Instantiating module "mschap" from file /etc/freeradius3/mods-enabled/mschap
rlm_mschap (mschap): using internal authentication
  # Instantiating module "expiration" from file /etc/freeradius3/mods-enabled/expiration
  # Instantiating module "files" from file /etc/freeradius3/mods-enabled/files
reading pairlist file /etc/freeradius3/mods-config/files/authorize
reading pairlist file /etc/freeradius3/mods-config/files/accounting
reading pairlist file /etc/freeradius3/mods-config/files/pre-proxy
  # Instantiating module "logintime" from file /etc/freeradius3/mods-enabled/logintime
  # Instantiating module "pap" from file /etc/freeradius3/mods-enabled/pap
  # Instantiating module "preprocess" from file /etc/freeradius3/mods-enabled/preprocess
reading pairlist file /etc/freeradius3/mods-config/preprocess/huntgroups
reading pairlist file /etc/freeradius3/mods-config/preprocess/hints
  # Instantiating module "IPASS" from file /etc/freeradius3/mods-enabled/realm
  # Instantiating module "suffix" from file /etc/freeradius3/mods-enabled/realm
  # Instantiating module "bangpath" from file /etc/freeradius3/mods-enabled/realm
  # Instantiating module "realmpercent" from file /etc/freeradius3/mods-enabled/realm
  # Instantiating module "ntdomain" from file /etc/freeradius3/mods-enabled/realm
  # Instantiating module "etc_passwd" from file /etc/freeradius3/mods-enabled/passwd
rlm_passwd: nfields: 3 keyfield 0(User-Name) listable: no
 } # modules
radiusd: #### Loading Virtual Servers ####
server { # from file /etc/freeradius3/radiusd.conf
} # server
server default { # from file /etc/freeradius3/sites-enabled/default
 # Loading authenticate {...}
Compiling Auth-Type PAP for attr Auth-Type
Compiling Auth-Type CHAP for attr Auth-Type
Compiling Auth-Type MS-CHAP for attr Auth-Type
 # Loading authorize {...}
Ignoring "sql" (see raddb/mods-available/README.rst)
Ignoring "ldap" (see raddb/mods-available/README.rst)
Compiling Autz-Type New-TLS-Connection for attr Autz-Type
 # Loading preacct {...}
 # Loading accounting {...}
 # Loading post-proxy {...}
 # Loading post-auth {...}
Compiling Post-Auth-Type REJECT for attr Post-Auth-Type
Compiling Post-Auth-Type Challenge for attr Post-Auth-Type
Compiling Post-Auth-Type Client-Lost for attr Post-Auth-Type
} # server default
server inner-tunnel { # from file /etc/freeradius3/sites-enabled/inner-tunnel
 # Loading authenticate {...}
Compiling Auth-Type PAP for attr Auth-Type
Compiling Auth-Type CHAP for attr Auth-Type
Compiling Auth-Type MS-CHAP for attr Auth-Type
 # Loading authorize {...}
 # Loading session {...}
 # Loading post-proxy {...}
 # Loading post-auth {...}
 # Skipping contents of 'if' as it is always 'false' -- /etc/freeradius3/sites-enabled/inner-tunnel:366
Compiling Post-Auth-Type REJECT for attr Post-Auth-Type
} # server inner-tunnel
radiusd: #### Opening IP addresses and Ports ####
listen {
        type = "auth"
        ipaddr = *
        port = 0
   limit {
        max_connections = 16
        lifetime = 0
        idle_timeout = 30
   }
}
listen {
        type = "acct"
        ipaddr = *
        port = 0
   limit {
        max_connections = 16
        lifetime = 0
        idle_timeout = 30
   }
}
listen {
        type = "auth"
        ipv6addr = ::
        port = 0
   limit {
        max_connections = 16
        lifetime = 0
        idle_timeout = 30
   }
}
listen {
        type = "acct"
        ipv6addr = ::
        port = 0
   limit {
        max_connections = 16
        lifetime = 0
        idle_timeout = 30
   }
}
listen {
        type = "auth"
        ipaddr = 127.0.0.1
        port = 18120
}
Listening on auth address * port 1812 bound to server default
Listening on acct address * port 1813 bound to server default
Listening on auth address :: port 1812 bound to server default
Listening on acct address :: port 1813 bound to server default
Listening on auth address 127.0.0.1 port 18120 bound to server inner-tunnel
Listening on proxy address * port 41793
Listening on proxy address :: port 43397
Ready to process requests
```

### CODE REVIEW

**Comment by esaaprillia** [comment]
@neheb

File: net/freeradius3/Makefile [hunk 1]
```diff
@@ -8,20 +8,19 @@
...
-PKG_HASH:=fdd476949f3c991c19f14ef7199b522e5204896d139c69946381dce9b8922941
+PKG_HASH:=1e75f5fc1961d9854d1cb3c6921612fbe2b9edb8ee508a5a7cbd69f1e7607115
 
 PKG_MAINTAINER:=
 PKG_LICENSE:=GPL-2.0
 PKG_LICENSE_FILES:=COPYRIGHT LICENSE
 PKG_CPE_ID:=cpe:/a:freeradius:freeradius
 
 PKG_BUILD_DIR:=$(BUILD_DIR)/freeradius-server-$(PKG_VERSION)
-PKG_BUILD_PARALLEL:=0
```
Comment (neheb) [inline net/freeradius3/Makefile:24]: you sure about this?

Comment (esaaprillia) [reply net/freeradius3/Makefile:24]: ya. 

https://downloads.openwrt.org/snapshots/faillogs/aarch64_generic/packages/freeradius3/compile.txt

**Comment by neheb** [comment]
What’s with the constant rebasing?

**Comment by esaaprillia** [comment]
> What’s with the constant rebasing?

There isn't any

**Comment by neheb** [comment]
Have you tested with GCC14?

**Comment by esaaprillia** [comment]
> Have you tested with GCC14?

is there any freeradius3 problem with gcc 14? I haven't tried building with gcc 14

**Comment by neheb** [comment]
That enum mismatch warning becomes and error in GCC 14.

**Comment by esaaprillia** [comment]
@neheb 

if that's the case maybe freeradius3 isn't in the snapshots package release

i see freeradius has released version 3.0.27
https://github.com/FreeRADIUS/freeradius-server/releases/tag/release_3_0_27

I've also created a patch for freeradius-krb5 to prevent cross-compiling tests

I haven't tested freeradius3 using gcc14 because openwrt hasn't released gcc14 yet
https://downloads.openwrt.org/snapshots/targets/bcm27xx/bcm2708/

File: net/freeradius3/Makefile (hunk 1 above)
Comment (neheb) [reply net/freeradius3/Makefile:24]: Please revert this change. I added this after I saw failures with make -j 31. I have no reason to believe that this issue was fixed.

**Comment by neheb** [comment]
Not happy that GCC14 is not fixed, but hopefully upstream fixes it.

**Comment by neheb** [comment]
```
/builder/shared-workdir/build/sdk/staging_dir/toolchain-arm_cortex-a8+vfpv3_gcc-13.3.0_musl_eabi/bin/../lib/gcc/arm-openwrt-linux-muslgnueabi/13.3.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ld.bfd: cannot find @LIBREADLINE@: No such file or directory
collect2: error: ld returned 1 exit status
```

**Comment by esaaprillia** [comment]
> ```
> /builder/shared-workdir/build/sdk/staging_dir/toolchain-arm_cortex-a8+vfpv3_gcc-13.3.0_musl_eabi/bin/../lib/gcc/arm-openwrt-linux-muslgnueabi/13.3.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ld.bfd: cannot find @LIBREADLINE@: No such file or directory
> collect2: error: ld returned 1 exit status
> ```

I'll try building and removing @LIBREADLINE@ I'll let you know later

**Comment by esaaprillia** [comment]
seems different

```
~/ko/freeradius-server-3.2.5 $ grep -r LIBREADLINE *
Make.inc.in:LIBREADLINE = @LIBREADLINE@
configure:LIBREADLINE
configure:    LIBREADLINE="$vl_cv_lib_readline"
configure:printf "%s\n" "#define HAVE_LIBREADLINE 1" >>confdefs.h
m4/vl_lib_readline.m4:    LIBREADLINE="$vl_cv_lib_readline"
m4/vl_lib_readline.m4:    AC_DEFINE(HAVE_LIBREADLINE, 1,
m4/vl_lib_readline.m4:  AC_SUBST(LIBREADLINE)
src/main/radmin.mk:TGT_LDLIBS   := $(LIBS) $(LIBREADLINE)
src/main/radmin.c:#ifdef HAVE_LIBREADLINE
src/main/radmin.c:#endif /* HAVE_LIBREADLINE */
src/include/autoconf.h.in:#undef HAVE_LIBREADLINE
```

```
~/ko/freeradius-server-3.0.27 $ grep -r LIBREADLINE *
Make.inc.in:LIBREADLINE = @LIBREADLINE@
acinclude.m4:    LIBREADLINE="$vl_cv_lib_readline"
acinclude.m4:    AC_DEFINE(HAVE_LIBREADLINE, 1,
acinclude.m4:  AC_SUBST(LIBREADLINE)
configure:LIBREADLINE
configure:    LIBREADLINE="$vl_cv_lib_readline"
configure:$as_echo "#define HAVE_LIBREADLINE 1" >>confdefs.h
src/main/radmin.mk:TGT_LDLIBS   := $(LIBS) $(LIBREADLINE)
src/main/radmin.c:#ifdef HAVE_LIBREADLINE
src/main/radmin.c:#endif /* HAVE_LIBREADLINE */
src/include/autoconf.h.in:#undef HAVE_LIBREADLINE
```

**Comment by neheb** [comment]
Variables need overriding as it doesn’t use pkgconfig.

**Comment by esaaprillia** [comment]
> Variables need overriding as it doesn’t use pkgconfig.

ac_cv_lib_readline=yes

**Comment by esaaprillia** [comment]
i will try to do that and update the freeradius-krb5 patch

**Comment by esaaprillia** [comment]
vl_cv_lib_readline

No

ac_cv_lib_readline

**Pull request merged**